tests.c

```diff
@@ -97,6 +97,110 @@ static char * test_lic0_invalid(){
     return 0;
 }
 
+/*
+*  A test where check_lic_2 should return true
+* @return 0 if the test passes, an error message otherwise
+*/
+static char * test_lic2_positive(){
+    NUMPOINTS = 6;
+    PARAMETERS.EPSILON = 0;
+    double local_X[NUMPOINTS];
+    double local_Y[NUMPOINTS];
+    X = local_X;
+    Y = local_Y;
+    X[0]=1; X[1]=2; X[2]=1; X[3]=4; X[4]=5; X[5]=6;
+    Y[0]=1; Y[1]=2; Y[2]=2; Y[3]=4; Y[4]=5; Y[5]=6;
+
+    mu_assert("The positive test failed!", check_lic_2() == true);
+
+    NUMPOINTS = 3;
+    double local_X1[NUMPOINTS];
+    double local_Y1[NUMPOINTS];
+    X = local_X1;
+    Y = local_Y1;
+    X[0]=4; X[1]=6; X[2]=5;
+    Y[0]=4; Y[1]=6; Y[2]=5;
+    mu_assert("The positive test failed!", check_lic_2() == true);
+  
+
+    NUMPOINTS = 3;
+    double local_X2[NUMPOINTS];
+    double local_Y2[NUMPOINTS];
+    X = local_X2;
+    Y = local_Y2;
+    PARAMETERS.EPSILON = PI/2;
+    X[0]=1; X[1]=2; X[2]=2;
+    Y[0]=1; Y[1]=2; Y[2]=1;
+
+    mu_assert("The positive test failed!", check_lic_2() == true);
+
+    return 0;
+}
+
+/*
+*  A test where check_lic_2 should return false
+* @return 0 if the test passes, an error message otherwise
+*/
+static char * test_lic2_negative(){
+    NUMPOINTS = 3;
+    PARAMETERS.EPSILON = PI;
+    double local_X[NUMPOINTS];
+    double local_Y[NUMPOINTS];
+    X = local_X;
+    Y = local_Y;
+    X[0]=4; X[1]=6; X[2]=5;
+    Y[0]=4; Y[1]=6; Y[2]=5;
+    mu_assert("The positive test failed!", check_lic_2() == false);
+
+    NUMPOINTS = 3;
+    PARAMETERS.EPSILON = PI/2;
+    double local_X1[NUMPOINTS];
+    double local_Y1[NUMPOINTS];
+    X = local_X1;
+    Y = local_Y1;
+    X[0]=4; X[1]=4; X[2]=5;
+    Y[0]=4; Y[1]=4; Y[2]=5;
+    mu_assert("The positive test failed!", check_lic_2() == false);
+
+    return 0;
+}
+
+/*
+*  A test where check_lic_2 should return false with an invalid input
+* @return 0 if the test passes, an error message otherwise
+*/
+static char * test_lic2_invalid(){
+    NUMPOINTS = 4;
+    PARAMETERS.EPSILON = 0;
+    double local_X[NUMPOINTS];
+    double local_Y[NUMPOINTS];
+    X = local_X;
+    Y = local_Y;
+    X = NULL;
+    Y = NULL;
+    mu_assert("The invalid test failed!", check_lic_0() == false);
+
+    NUMPOINTS = 4;
+    PARAMETERS.EPSILON = PI;
+    double local_X1[NUMPOINTS];
+    double local_Y1[NUMPOINTS];
+    X = local_X1;
+    Y = local_Y1;
+    X[0]=1; X[1]=2; X[2]=2;
+    Y[0]=1; Y[1]=2; Y[2]=1;
+    mu_assert("The invalid test failed!", check_lic_0() == false);

```

Comment:
I am not sure how this can be checked in C. The test makes sense in a language like Java but if you try to access X[3] when it's not set to a value you just get a segmentation fault in C. (You wouldn't get a segmentation fault in this case, but in general you can't be sure that X[i] doesn't point to a location you don't have access to)

Reply:
Yes, my bad it was supposed to be 3. Anyway I was testing here with epsilon not in the interval [0;Pi[.